substrate/frame/nomination-pools/benchmarking/src/mock.rs

```diff
@@ -166,7 +166,7 @@ impl pallet_nomination_pools::Config for Runtime {
 	type RewardCounter = FixedU128;
 	type BalanceToU256 = BalanceToU256;
 	type U256ToBalance = U256ToBalance;
-	type Staking = Staking;
+	type StakeAdapter = pallet_nomination_pools::adapter::TransferStake<Self, Staking>;
```

Comment:
Assuming DelegateStake will be worse case, should I becch against that or is there a good way to bench with different configurations for different runtime?

Reply:
For the `SubstrateWeight` I think it's ok to bench against whatever adapter. Actual runtimes will bench this pallet again with their specific runtime config for `StakeAdapter`, so as long as the benchmark functions hit the right codepaths regardless of adapters, it's ok.